### PR TITLE
DOC Enhance the install and contributing sections

### PR DESCRIPTION
**Pull request by Vincent-Maladiere**
**Reference**: https://github.com/skrub-data/skrub/issues/1071

**What does this PR propose?**

- Clarifying the redirect from the contributing page to the "Install > From source" section. Note that clicking on the link on the current doc takes you to the pip section, not to the "From source" section.
- Adding the lint and test dependencies to the "From source" guideline
- Removing the table of contents at the beginning of the document, because:
  - It is redundant: the info is displayed on the right panel
  - It takes some vertical place
  - It turns every header into a link, which is confusing, and brings you back to the table of contents on click (see [this comment](https://github.com/skrub-data/skrub/issues/1071#issuecomment-2349570135)).
- Mention Discord as a place to discuss and ask questions
- Some rephrasing and typos fixing.

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -49,59 +49,98 @@ Install
...
-Advanced usage, for contributors
---------------------------------
+Advanced Usage for Contributors
+-------------------------------
+
+1. Fork the project
+'''''''''''''''''''
+
+To contribute to the project, you first need to
+`fork skrub on Github <https://github.com/skrub-data/skrub/fork>`_.
```
Comment (TheooJ) [inline doc/install.rst:67]:
```suggestion
`fork skrub on GitHub <https://github.com/skrub-data/skrub/fork>`_.
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -3,118 +3,111 @@ Contributing to skrub
...
-Also, if scientific literature doesn't scare you, we greatly
-encourage you to read the two following papers:
+To understand the purpose and goals behind skrub, please read our
+`vision statement. <https://skrub-data.org/stable/vision.html>`_
 
-- `Similarity encoding for learning
-  with dirty categorical variables <https://hal.inria.fr/hal-01806175>`__
-- `Encoding high-cardinality string categorical
-  variables <https://hal.inria.fr/hal-02171256v4>`__.
+If you're interested in the research behind skrub,
```
Comment (TheooJ) [inline CONTRIBUTING.rst:23]: I'm not sure I'd mention this in the "What to know before you begin" section, maybe in a "to go further" somewhere else ?

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -3,118 +3,111 @@ Contributing to skrub
...
-    let us know there are several users affected by this issue.
-  - **Issue has been closed**:
+1. **Check if an issue already exists**
+   by searching the `GitHub issues <https://github.com/skrub-data/skrub/issues?q=is%3Aissue>`_
 
-    - **By a merged pull request** (1) update your skrub version,
-      or (2) the fix has not been released yet.
-    - **Without pull request**, there might be a ``wontfix`` label, and/or a reason at the bottom of the conversation.
+   - If **open**, leave a 👍 on the original message to signal that others are affected.
+   - If closed, check for one of the following:
```
Comment (TheooJ) [inline CONTRIBUTING.rst:44]: Weird, this line appears in bold in the rendered docs, but I don't see why

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -3,118 +3,111 @@ Contributing to skrub
...
-  developer can start working on it. Note however that we cannot guarantee how much time
-  it will take to implement the change.
 
 If the enhancement is refused
 '''''''''''''''''''''''''''''
 
-There are specific incentives behind skrub. While most enhancement
-ideas are good, they don't always fit in the context of the library.
+Although many ideas are great, not all will align with the objectives
+of the skrub library.
```
Comment (TheooJ) [inline CONTRIBUTING.rst:103]:
```suggestion
of skrub.
```

?

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -124,62 +117,51 @@ Writing your first Pull Request
...
    git switch -c branch_name
 
-Finally, install the dependencies by heading to the `installation process <https://skrub-data.org/stable/install.html#advanced-usage-for-contributors>`__,
-advanced usage section.
-
-Implementation
-^^^^^^^^^^^^^^
 
-There are a few specific project goals to keep in mind:
+.. _implementation guidelines:
```
Comment (TheooJ) [inline CONTRIBUTING.rst:137]: I'd use
```suggestion
.. _implementation-guidelines:
```

here and in install.rst

**Review by jeromedockes — APPROVED** [review]
thanks!

**Review by TheooJ — APPROVED** [review]
I think the contributing page is much clearer this way. +1 for all the additions

File: CONTRIBUTING.rst (hunk 2 above)
Comment (Vincent-Maladiere) [reply CONTRIBUTING.rst:23]: It used to be there, so I didn't challenge that idea. While I think highlighting the research background of the project is helpful, I also think the GapEncoder and MinHashEncoder are slightly less central to skrub than they used to be. This needs to be discussed with the team.

File: CONTRIBUTING.rst (hunk 3 above)
Comment (Vincent-Maladiere) [reply CONTRIBUTING.rst:44]: Yes, not sure why either

File: CONTRIBUTING.rst (hunk 5 above)
Comment (Vincent-Maladiere) [reply CONTRIBUTING.rst:137]: I use this small tweak to put the anchor as is in the text, see: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#use-a-custom-anchor

Comment (TheooJ) [reply CONTRIBUTING.rst:137]: Yes, I was wondering if we should use ``-`` between words when creating anchors, like in ``.. _writing-your-first-pull-request:`` L112

**Comment by jeromedockes** [comment]
very nice, thanks a lot!

**Pull request merged**